### ISSUE TITLE
[Audit] Dropdown

### ISSUE THREAD

**Issue by joshblack** [carbon-design-system/carbon#1955]
## Dropdown

- [ ] Should dropdown be using listbox styles? Should we pull out listbox into a set of tools to apply styles instead of shared styles for each?

- [x] Should use `body-short-01` for placeholder text and value

- [x] What should focus be for error state? cc @laurenmrice Currently it's a red border

- [ ] All dropdown versions should be pulling the 16 icon. I think this is the old icon.

![Screen Shot 2019-03-11 at 10.08.58 AM.png](https://images.zenhubusercontent.com/5bbba771387e41362246f3dc/c1df16e3-764b-4f46-95d8-f011c2fcea62)

## Inline Dropdown

- [x] Should use `body-short-01` for text

- [x] Bounding box should have only `8px` of padding on left hand side

Expected:

![image](https://user-images.githubusercontent.com/3901764/53582903-13f3b900-3b46-11e9-91c1-8cd0fd9ca25a.png)

Actual:

![image](https://user-images.githubusercontent.com/3901764/53582921-1eae4e00-3b46-11e9-94d4-faf201578901.png)

- [ ] Width of control seems off from spec

Expected:

![image](https://user-images.githubusercontent.com/3901764/53582960-31c11e00-3b46-11e9-9ec1-5da076548a08.png)

Actual:

![screen shot 2019-02-28 at 10 46 59 am](https://user-images.githubusercontent.com/3901764/53582968-3554a500-3b46-11e9-8d87-896f5abb7fb6.png)

## Filterable Dropdown (Combobox), MultiSelects, etc.

These all use `list-box` and will most likely share updates

- [x] Should use `body-short-01` for text
- [x] Disabled -> hover seems to still have hover color
- [ ] Need to verify in React repo for functionality

**Comment by laurenmrice** [comment]
@joshblack For dropdown focus for error state: The error state should have a 2px red border (support-01, #DA1E28). Are you saying that it should also get a focus state applied on top of the error state? If so, the focus should be 2px blue border (focus, #0062FF)

**Comment by emyarod** [comment]
depends on https://github.com/IBM/carbon-components/pull/1987

**Comment by emyarod** [comment]
depends on https://github.com/IBM/carbon-components/pull/1987

I believe some of the points have been addressed already in https://github.com/IBM/carbon-components/pull/1793 https://github.com/IBM/carbon-components/pull/1896 https://github.com/IBM/carbon-components/pull/1898

> * Bounding box should have only 8px of padding on left hand side

so are we undoing the requested feedback from [this audit](https://github.com/IBM/carbon-components/issues/1749)

just to clarify: are the points for combobox and multiselect the entire audit feedback for those components? or should those be in a separate ticket?

**Comment by asudoh** [comment]
@joshblack @laurenmrice Any thoughts on what @emyarod mentioned above? Thanks!

**Comment by laurenmrice** [comment]
- #1896, #1989, #1793 are closed issues and are good.
- Just reviewed with changes for open issue #1987.
- For the #1749 audit issue, those listed changes for dropdown are all still valid for things that need to be fixed if they are not all ready implemented. Can I have the current link of dropdown to check if those fixes were made? 
- For Combo-box and Multi-select I would need to see the latest version of those to determine if that is all that needs to be changed.

**Comment by emyarod** [comment]
@laurenmrice you can view the dropdown example for this ticket here https://deploy-preview-1994--the-carbon-components.netlify.com/ (the combobox and multiselect components can be viewed there too, no changes have been made to those yet)

**Comment by laurenmrice** [comment]
**Changes:**
Things that still need to be changed that are mentioned in #1749:
   - [x] For normal dropdowns: Update padding to 16px on left/right (icon should be 16px away from right), vertically centered text. 
   - [x] The Option text in the drawer by default should be text-02 color (#565656). The Option text turns text-01 (#171717) when it gets a hover or selection background.
   - [x] When open, label should remain color token field-01 (#f3f3f3.) It should not get a select color. This also happens for the error state version.
   - [x] When selecting a new option, the background color should go to color token selected-ui (#dcdcdc). Right now it looks too dark.
![Screen Shot 2019-03-07 at 2.27.31 PM.png](https://images.zenhubusercontent.com/5bbba771387e41362246f3dc/34883ab9-f236-4585-9bae-53562627156a)
   - [x] The focus outline of the first and last option in the dropdown when it is open needs to be flush against the sides of the menu. Right now it is hanging.

- [x] Variation of dropdown with a label and helper text is missing in netlify link ?
- [x] Disabled dropdown should not get a border bottom.

**Comment by emyarod** [comment]
> Update padding to 16px on left/right (icon should be 16px away from right), vertically centered text

so this point contradicts what we have written in the current ticket (#1955) unless I am misunderstanding "Bounding box should have only 8px of padding on left hand side". If what we have from #1749 stands then I don't need to make any changes here since #1749's changes are already merged in

but I can make the remaining changes, thanks for the updates! The netlify link only shows my progress up so far but I needed clarification on several points, and I'm also blocked by https://github.com/IBM/carbon-components/pull/1987 which is why the inline dropdowns still look off at the moment

**Comment by laurenmrice** [comment]
The 8px padding for bounding box only applies for the inline dropdown. I was referring to the normal dropdowns when i mentioned 16px
